8.0 Add support for partner bank matching on invoice update

alexis-via opened the pull request:
Add support for partner bank matching on invoice update. Before, this feature was only supported on invoice creation. I developed this new feature this afternoon for my demo at my Odoo experience conference.

alexis-via committed:
Add support for partner bank matching on invoice update (before, it was only supported on invoice creation)

pedrobaeza commented on base_business_document_import/models/business_document_import.py:
```diff
@@ -238,6 +238,36 @@ def _match_shipping_partner(self, shipping_dict, partner, chatter_msg):
+        iban = iban.replace(' ', '')
+        rpbo = self.env['res.partner.bank']
+        self._cr.execute(
+            """SELECT id FROM res_partner_bank
```
Why make this with SQL?

alexis-via replied:
I need the replace() method in SQL because the IBAN can have spaces in res_partner_bank. In v9, this problem is solved by the acc_number_sanitized field, so I will remove this SQL request in v9 and use the ORM instead.

pedrobaeza replied:
OK

pedrobaeza commented on base_business_document_import/models/business_document_import.py:
```diff
@@ -238,6 +238,36 @@ def _match_shipping_partner(self, shipping_dict, partner, chatter_msg):
+                'acc_number': iban,
+                'bank_bic': bic,
+                })
+            chatter_msg.append(_(
```
You have to modify the translations due to this file change I think

alexis-via replied:
Translations are handled automatically via transifex, no ?

pedrobaeza replied:
Yes, but you will lose any of the already present translations. It's just to check and modify translation files to see the comment line about the code that contains this sentence.

alexis-via replied:
There are no translation for this module so far.

pedrobaeza replied:
OK

pedrobaeza commented on base_business_document_import/models/business_document_import.py:
```diff
@@ -238,6 +238,36 @@ def _match_shipping_partner(self, shipping_dict, partner, chatter_msg):
 
     @api.model
+    def _match_partner_bank(
+            self, partner, iban, bic, chatter_msg, create_if_not_found=False):
```
To make it more tolerant, put `chatter_msg` as optional.

alexis-via replied:
Mmm, it is that way in all the other methods of this module.

pedrobaeza replied:
But you can start with this one being more tolerant :wink: for reusing it in models without chatter.

alexis-via replied:
When I'll change that, I'll change it everywhere ; I don't want to do it partially.

pedrobaeza replied:
This is a good moment for that :wink:, but I know that you're not going to do it, hehe

alexis-via committed:
Update README.rst: migration to the edi project

I didn't update sale_commercial_partner because this module will have to be moved to sale-workflow

pedrobaeza reviewed (approved):
Squashing on merging

pedrobaeza commented:
@alexis-via, what are the plans for bringing this to v9?

alexis-via commented:
I'll port it to v10 in December 2016 or January 2017 (when my customers will go live on v10) ; I don't want to start the port too early because I still have things that I want to change on v8 and I would like to avoid too much backporting work. The 2 points I want to address are:
- the support of sale order import when your products have a price-include=True taxes
- add support for importing an invoice from a foreign country with VAT that you can't get back

I don't have any plans for v9, but my port to v10 should be compatible with v9 I guess.

pedrobaeza commented:
OK, thanks for sharing your plans. I'm not sure how to act because we need it on v9, but only for commodity (not mandatory requirement), but we want it now. I'll tell you later.
